src/java/org/apache/cassandra/service/consensus/migration/ConsensusMigrationMutationHelper.java

```diff
@@ -0,0 +1,399 @@
+    {
+        return mutateWithAccordAsync(ImmutableList.of(mutation), consistencyLevel, queryStartNanoTime);
+    }
+
+
+    public static Pair<TxnId, Future<TxnResult>> mutateWithAccordAsync(Collection<? extends IMutation> mutations, @Nullable ConsistencyLevel consistencyLevel, long queryStartNanoTime)
+    {
+        int fragmentIndex = 0;
+        List<TxnWrite.Fragment> fragments = new ArrayList<>(mutations.size());
+        List<PartitionKey> partitionKeys = new ArrayList<>(mutations.size());
+        for (IMutation mutation : mutations)
+        {
+            for (PartitionUpdate update : mutation.getPartitionUpdates())
+            {
+                PartitionKey pk = PartitionKey.of(update);
+                partitionKeys.add(pk);
+                fragments.add(new TxnWrite.Fragment(PartitionKey.of(update), fragmentIndex++, update, TxnReferenceOperations.empty()));
+            }
+        }
+        // Potentially ignore commit consistency level if the strategy specifies accord and not migration
+        ConsistencyLevel clForCommit = consistencyLevelForCommit(mutations, consistencyLevel);
+        TxnUpdate update = new TxnUpdate(fragments, TxnCondition.none(), clForCommit, true);
+        Txn.InMemory txn = new Txn.InMemory(Keys.of(partitionKeys), TxnRead.EMPTY, TxnQuery.NONE, update);
+        IAccordService accordService = AccordService.instance();
+        return accordService.coordinateAsync(txn, consistencyLevel, queryStartNanoTime);
+    }
+
+    public static void dispatchMutationsWithRetryOnDifferentSystem(List<? extends IMutation> mutations, ConsistencyLevel consistencyLevel, long queryStartNanoTime)
+    {
+        while (true)
+        {
+            try
+            {
+                // Keep attempting to route the mutations until they all succeed on the correct system
+                // TODO (review): Metrics are going to double count when we split between the two systems
+                // but they will be split across different accord/non-accord metrics
+                Pair<List<IMutation>, List<IMutation>> accordAndNormal = splitMutationsIntoAccordAndNormal((List<IMutation>)mutations);
+                List<? extends IMutation> accordMutations = accordAndNormal.left;
+                // TODO this can race with migration to/from Accord and Accord doesn't know what range we are referring to or it just hasn't updated to the epoch we are aware of
+                // client side retry can still succeed but the error could be exposed
+                Pair<TxnId, Future<TxnResult>> accordResult = accordMutations != null ? mutateWithAccordAsync(accordMutations, consistencyLevel, queryStartNanoTime) : null;
+                List<? extends IMutation> normalMutations = accordAndNormal.right;
+
+                Throwable failure = null;
+                try
+                {
+                    if (normalMutations != null)
```

Comment:
I do it this way just because I don't normally ask the question "does this code throw", which in this case is `if (normalMutations != null)` just because putting something in a `try` block is free and then later changes that add/remove exceptions still get whatever the intended exception handling is.

So the example here would be if `normalMutations != null` were replaced with something more complex in the future that could NPE or `ClassCastException` it would still be handled.

The outer try block in this loop doesn't handle all exception types, and we can't rely on it because there are two sources of exceptions and we always want to get both (regular mutation path, accord path) before going through the loop again.

TL;DR I would argue that when using `try/catch/finally` the default approach should be to have everything possible inside the the `try` block since the cost of failing to handle exceptions is > than the cost of the extra indentation.